pkg/driver/deployment/knative.go

```diff
@@ -83,11 +87,63 @@ func knativeDeploySingleFunction(function *common.Function, yamlPath string, isP
 		// second, then round to an integer as that is what the knative config expects
 	}
 
+	// Default is the function name, but if the function is a vSwarm function, the proxy name is used
+	proxyName := function.Name
+
+	if vSwarm {
+		// Read and unmarshal the mapper output file into a map
+
+		mapperFile, err := os.ReadFile("tools/mapper/output.json")
+		if err != nil {
+			log.Warn("No mapper output file")
+		}
+		var mapperOutput map[string]map[string]string // HashFunction mapped to vSwarm function yaml.
+		err = json.Unmarshal(mapperFile, &mapperOutput)
+		if err != nil {
+			log.Warn("Failed to unmarshal mapper output file")
+		}
+
+		// Find the proxy name for the function
+		proxyName = mapperOutput[function.HashFunction]["proxy-function"]
+		log.Info("Proxy name: ", proxyName)
+		// Read the deployment info file for yaml locations and predeployment commands if any
+
+		deploymentInfoFile, err := os.ReadFile(yamlPath + "deploy_info.json")
+		if err != nil {
+			log.Warn("No deployment info file")
+		}
+
+		var deploymentInfo map[string]map[string]interface{}
```

Comment:
Why do you use this data structure? Can you use sensible structs for that?

Reply:
This data structure is to unmarshal the json file that has the predeployment commands that are required for deploying functions that require databases.

Reply:
Can you use structure of structures instead of map of maps?

Reply:
Should I define a new struct for this? I thought this is a pretty simple interface to unmarshal into and that adding a specific struct for this would complicate things.